### PR TITLE
Storage tests new

### PR DESCRIPTION
**Pull request by RaoulSchaffranek**
This PR replaces #1310

### CODE REVIEW

**Comment by ehildenb** [comment]
@RaoulSchaffranek for the new tests, you can add them to `tests/failing-symbolic.java` (so add `tests/specs/examples/storage-spec.k` to that list), so taht you don't have to worry about getting it working on the Java backend.

**Comment by ehildenb** [comment]
Nice, looks really good.

**Comment by ehildenb** [comment]
Did you profile the proofs? How long are they taking? If it's more than 10m we should probably add them to `tests/slow-symbolic.haskell` to avoid running them on CI.

**Comment by ehildenb** [comment]
Also, it appears you need to mark this test as failing on the java backend, by listing it in `tests/failing-symbolic.java`.

**Comment by RaoulSchaffranek** [comment]
Thanks, the tests take 30 min on my machine and 15 minutes on Andrejs. I added a line to the `haskell.slow` file and `failing-symbolic.java` respectively.

**Review by ehildenb — APPROVED** [review]

File: tests/specs/examples/storage-spec.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+          <output>     .ByteArray                       </output>
+          <statusCode> _          => EVMC_SUCCESS       </statusCode>
+
+          <account>
+            <acctID> ACCTID </acctID>
+            <storage> ACCT_STORAGE => ACCT_STORAGE [ MYBOOL_KEY <- NEW_STORAGE_CONTENT ] </storage>
+            ...
+          </account>
+
+          <refund> _ => ?_ </refund>
```
Comment (RaoulSchaffranek) [inline tests/specs/examples/storage-spec.md:136]: @JuanCoRo

**Pull request merged**